docs/cookbook/creating-webhooks.md

```diff
@@ -0,0 +1,93 @@
+# Creating webhooks
+
+Webhooks allow applications to communicate with each other in real time by sending HTTP requests when certain events or triggers occur. This can allow your site to automatically send notifications when publishing a document or receive updates from payment portals, for example. This short recipe will demonstrate how to set up webhooks in your Apostrophe project.
+
+## Outgoing webhooks
+
+There are a wide variety of applications and services that expose webhooks for the transfer of information from your site. For example, you can post messages in specific Slack or Discord channels using a webhook. Apostrophe [handlers](/reference/module-api/module-overview.html#handlers-self) are able to capture a number of built-in [server-side events](../reference/server-events.md) and you also have the option of adding custom events emitted from your own modules. These events are perfect for automatically triggering the delivery of data about an event from your site to a 3rd party.
+
+<AposCodeBlock>
+
+```javascript
+handlers(self) {
+  return {
+    'afterPublish': {
+      async sendWebhook(req, data) {
+        // data sent by the afterPublish event includes the document data 
+        // and whether it is being published for the first time
+        // https://v3.docs.apostrophecms.org/reference/server-events.html#afterpublish
+        if (!data.firstTime) {
+          return;
+        }
+
+        await self.apos.http.post('https://slack-webhook-url-here', {
+          body: {
+            text: 'do something cool with the data'
```

Comment:
How about we show how to send the title of the document as part of this? Seems not difficult to do

Reply:
Yeah, sure.
I also wasn't sure about not setting a `const` equal to the response that is returned from the `POST`. Do you think people are going to be asking the followup about knowing how they got a `400`?